pennylane/ops/functions/dot.py

```diff
@@ -136,6 +137,12 @@ def dot(
     if len(coeffs) == 0 and len(ops) == 0:
         raise ValueError("Cannot compute the dot product of an empty sequence.")
 
+    for t in (Operator, PauliWord, PauliSentence):
+        if isinstance(ops, t):
+            raise ValueError(
+                f"ops must be an Iterable of {t.__name__}'s, not a {t.__name__} itself."
+            )
```

Comment:
```suggestion
    if isinstance(ops, (Operator, PauliWord, PauliSentence)):
        raise ValueError(
            f"ops must be an Iterable of {t := type(ops)}'s, not a {t} itself."
        )
```

not sure if this works with the walrus

Reply:
Also tried something like that. but I thought:
```
"ops must be an Iterable of <class 'pennylane.ops.op_math.prod.Prod'>'s, not a <class 'pennylane.ops.op_math.prod.Prod'> itself."
```
wasn't what I was going for.  I wanted it to say `Operator` instead of `Prod`.